### PR TITLE
SSEMR-335 Adding filter mechanism.

### PR DESCRIPTION
**Pull request by Michaelndula**
## Release Notes

- **New Features**
	- Introduced a method to calculate patient age based on birthdate.
	- Added a utility class for applying filters based on patient attributes, enhancing data retrieval.
	- Implemented filtering capabilities in various patient data retrieval methods for more precise results.
	- Enhanced methods for retrieving active and new patients to include filtering based on specified criteria.

- **Bug Fixes**
	- Streamlined error handling in patient pagination, allowing for a more flexible approach.

- **Documentation**
	- Updated method signatures to reflect new filtering parameters in relevant controllers.

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/ssemrws/constants/SharedConstants.java [hunk 1]
```diff
@@ -227,6 +227,13 @@ public static long calculateAge(Date birthdate) {
 		return (currentDate.getTime() - birthdate.getTime()) / (1000L * 60 * 60 * 24 * 365);
 	}
 	
+	public static int getPatientAge(Patient patient) {
+		Date birthdate = patient.getBirthdate();
```
Comment (nelsonkimaiga) [inline omod/src/main/java/org/openmrs/module/ssemrws/constants/SharedConstants.java:231]: Why the decision to work with `Date` as opposed to `LocalDate`?

File: omod/src/main/java/org/openmrs/module/ssemrws/web/constants/FilterUtility.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+public class FilterUtility {
+	
+	public static boolean applyFilter(Patient patient, SSEMRWebServicesController.filterCategory filterCategory,
+	        Date endDate) {
+		if (filterCategory == null) {
+			return true;
+		}
+		switch (filterCategory) {
+			case CHILDREN_ADOLESCENTS:
+				return getPatientAge(patient) <= 19;
```
Comment (nelsonkimaiga) [inline omod/src/main/java/org/openmrs/module/ssemrws/web/constants/FilterUtility.java:25]: @Michaelndula if 19 acts as an age limit, you can extract it to a constant variable for easy modification in the future

File: omod/src/main/java/org/openmrs/module/ssemrws/constants/SharedConstants.java (hunk 1 above)
Comment (Michaelndula) [reply omod/src/main/java/org/openmrs/module/ssemrws/constants/SharedConstants.java]: @nelsonkimaiga I have resolved this.

File: omod/src/main/java/org/openmrs/module/ssemrws/web/constants/FilterUtility.java (hunk 2 above)
Comment (Michaelndula) [reply omod/src/main/java/org/openmrs/module/ssemrws/web/constants/FilterUtility.java]: @nelsonkimaiga I have resolved this as well.

**Pull request merged**